tests/components/cast/test_media_player.py

```diff
@@ -173,7 +184,17 @@ def discover_chromecast(service_name: str, info: ChromecastInfo) -> None:
         assert get_chromecast.call_count == 1
         assert cast.CastStatusListener.call_count == 1
         entity = cast.CastStatusListener.call_args[0][0]
-        return chromecast, entity
+        chromecast.cast_status_listener = entity._status_listener.new_cast_status
```

Comment:
Why do we need to set these attributes on the chromecast? Isn't that done during integration setup?

Reply:
It's used in a patch. Are these attributes never set in the code under test?

We should use the mock to get access to the callbacks we need to call to trigger behavior we want to test. We don't want to interact with any integration details in the test.

Reply:
Yes, you're right: the callbacks are in fact set by the cast integration, so it should be possible to dig them up from calls made on the mock.